boot/bootchain.go

```diff
@@ -105,30 +105,14 @@ func stringListsLess(sl1, sl2 []string) bool {
 	return false
 }
 
-func toPredictableBootAsset(b *bootAsset) *bootAsset {
-	if b == nil {
-		return nil
-	}
-	newB := *b
-	if b.Hashes != nil {
-		newB.Hashes = make([]string, len(b.Hashes))
-		copy(newB.Hashes, b.Hashes)
-		sort.Strings(newB.Hashes)
-	}
-	return &newB
-}
-
 func toPredictableBootChain(b *bootChain) *bootChain {
 	if b == nil {
 		return nil
 	}
 	newB := *b
-	if b.AssetChain != nil {
-		newB.AssetChain = make([]bootAsset, len(b.AssetChain))
-		for i := range b.AssetChain {
-			newB.AssetChain[i] = *toPredictableBootAsset(&b.AssetChain[i])
-		}
-	}
+	// AssetChain is sorted list (by boot order) of sorted list (old to new asset).
+	// So it is already predictable and we can keep it the way it is.
+	newB.AssetChain = b.AssetChain
```

Comment:
Maybe it would make sense to make a deep copy as we do anyway for `KernelCmdlines`

Reply:
Yes, it would need to copy `Hashes` as well - I guess it is fine, I'm just slightly worried that one of the slices changes under our feet for the returned object, although most probably that is not the case. Otherwise, LGTM.